ADO.NET/OAuthSampleApp/README.md

```diff
@@ -0,0 +1,29 @@
+## Overview
+
+The Vertica ADO.NET driver now allows connecting through OAuth access tokens. 
+
+This sample application was built in .NET 6.0 and can be run in .NET Framework 4.6+ or .NET Core 3.1+.
+
+# Prerequisites
+
+For information on installing .NET, starting a Vertica docker container, and how to reference a NuGet package, see the README.md in the ADO.NET base directory.
+
+# Using OAuth to login
+
+You must first configure the Vertica server for OAuth as described here: https://docs.vertica.com/24.1.x/en/security-and-authentication/client-authentication/oauth-2-0-authentication/configuring-oauth-authentication/.
+The steps can also be done programatically as shown in the SetUp method.
+
+The doc describes how to get an access token through Curl, but the sample app has functions for getting access and refresh tokens.
+The authentication steps are:
+1. Use the OAuth credentials from the app.config to set an access and refresh token.
+2. Attempt to login with the access token. If it's expired, use the refresh token to get a new access token and login with that.
+3. If the refresh token is expired, repeat step 1.
+
+This app stores confidential information like the client secret in the app.config file for ease of use. Use proper secret management tools in production.
+
+Note: The token url should be in the form of "http://192.168.0.255:8080/realms/test/protocol/openid-connect/token/"
```

Comment:
This is keycloak specific, we may want to find a way to generalize in the future

Reply:
I added a note